### DISCUSSION TITLE
Cli based import for large analyses

### DISCUSSION THREAD

**Discussion by K8Sudo** [monarc-project/MonarcAppFO#406] [Ideas]
For large risk analyses of 150 MB file size and more, the import process takes a very long time even on high-performance hardware. For this analysis, it would be very helpful to be able to import using cli.

**Comment by ruslanbaidan** [comment]
We worked on reducing the import process time in the Monarc UI.
It was improved, but the further work is planned. That's why [the ticket](https://github.com/monarc-project/MonarcAppFO/issues/248) is still open.
If we are talking about cli import, this requires access to the server by a user, which can't be provided.
We can consider a solution with a background process run, which can be executed from the UI and notify frontend when it is finished.

**Reply by K8Sudo** [reply]
I noticed the import improvement and also saw that the ticket is still open. I thought that an import via CLI might be more quickly implementable than a background job via UI. The import option via CLI would also have been more about an administrative provision of exports, rather than a user activity. The background job still sounds like a very good idea though and I'm looking forward to when the feature is ready. Currently running imports with 150MB json file needs 13-18h.

Thanks for the info on the upcoming plan

**Reply by ruslanbaidan** [reply]
Thank you for sharing your idea.
Basically the CLI export, as you call it, does not exclude the scenario with UI and can help to speed up a little and avoid the user waiting time also errors occurrence due to request/response timeout, memory limit etc.
The CLI process can be executed by request from frontend and it's status validated with quick status requests.
When the export process is finished the file can be available to download.

**Reply by ruslanbaidan** [reply]
Hi @KonzeptAcht ,
The functionality is implemented within v2.12.6.
To enable it, there is a flag [here](https://github.com/monarc-project/MonarcAppFO/blob/master/config/autoload/local.php.dist#L122) to set to true. The files storage directory has to be created as well (`uploadFolder`).
The same is in the ansible [config template](https://github.com/monarc-project/ansible-ubuntu/blob/master/playbook/monarcfo/templates/client_config.php.j2#L109) if the ansible deployment is used. It can be now set via the clients creation on the BackOffice side UI.
